CrashReporter/file_uploader.cpp

```diff
@@ -19,23 +25,54 @@ FileUploader::~FileUploader()
 
 bool FileUploader::uploadMinidump()
 {
+#ifdef Q_OS_WIN
     typedef std::basic_string<wchar_t> wstring;
     std::map<wstring, wstring> files;
     std::map<wstring, wstring> parameters;
     parameters[L"Logs"] =  QString(additionalData).toStdWString();
     files[L"upload_file_minidump"] = dmpFilePath.toStdWString();
     qDebug() << "uploading file: " << dmpFilePath;
 
-    return google_breakpad::HTTPUpload::SendRequest(
+    google_breakpad::HTTPUpload::SendRequest(

```

Comment:
@rish9511 SendRequest might be an asynchronous call. So before even completing SendRequest, emit uploadDone() might get executed in that case.

Reply:
@sahil21  it's a synchronous request